packages/clay-dropdown/src/ClayDropdownBase.js

```diff
@@ -214,92 +149,80 @@ class ClayDropdownBase extends ClayComponent {
 	}
 
 	/**
-	 * Sets the next item in the list as active.
-	 * @param {Boolean} reverse
-	 * @private
+	 * Handles document click in order to hide menu.
+	 * @param {!Event} event
+	 * @protected
 	 */
-	_setNextActive(reverse) {
-		const items = flatten(this.items, true);
-		let activeSubIndex = -1;
-
-		const activeIndex = items.findIndex(item => {
-			const subIndex = item.findIndex(item => item.active);
-
-			if (subIndex !== -1) {
-				activeSubIndex = subIndex;
-
-				return true;
-			}
-		});
-
-		const {active, subActive} = this._getNextIndexes(
-			items,
-			activeIndex,
-			activeSubIndex,
-			reverse
-		);
-
-		if (items[activeIndex] && items[activeIndex][activeSubIndex]) {
-			items[activeIndex][activeSubIndex].active = false;
+	_handleDocClick(event) {
+		if (
+			this.element.contains(event.target) ||
+			(this.refs.portal &&
+				this.refs.portal.element.contains(event.target))
+		) {
+			return;
 		}
-
-		items[active][subActive].active = true;
-
-		this.items = this._assembleFromFlattenedGroups(items);
-	}
-
-	/**
-	 * Reassembles flattened items into the correct items structure.
-	 * This method is used in conjunction with `flatten(..., true)`
-	 * @param {Array} flattenedItems
-	 * @private
-	 * @return {Array}
-	 */
-	_assembleFromFlattenedGroups(flattenedItems) {
-		return this.items.map((item, i) => {
-			if (item.items) {
-				item.items = flattenedItems[i];
-			} else {
-				item = flattenedItems[i][0];
-			}
-
-			return item;
-		});
+		this.toggle();
 	}
 
 	/**
-	 * Handle key events.
+	 * Handles footer button click.
 	 * @param {!Event} event
-	 * @private
+	 * @protected
 	 */
-	_handleKeyNavigation(event) {
-		if (KEY_NAVIGATION.includes(event.key)) {
-			event.preventDefault();
-
-			if (event.key === KEY_DOWN || event.key === KEY_UP) {
-				this._setNextActive(event.key === KEY_UP);
-			} else if (event.key === KEY_ENTER) {
-				this._handleEnterKey(event);
-			}
-		}
+	_handleButtonClick(event) {
+		this.emit('buttonClicked', event);
 	}
 
 	/**
-	 * Continues the propagation of the enter key event
+	 * Continues the propagation of the item clicked event
 	 * @param {!Event} event
 	 * @protected
 	 * @return {Boolean} If the event has been prevented or not.
 	 */
-	_handleEnterKey(event) {
+	_handleItemClick(event) {
+		const element = event.delegateTarget;
+		const elementIndex = this._getDropdownItemIndex(element);
+		const flatenItems = flatten(this.items);
+		const item = flatenItems[elementIndex];
+
 		return !this.emit({
 			data: {
-				item: flatten(this.items).find(item => item.active),
+				item: item,
 			},
 			name: 'itemClicked',
 			originalEvent: event,
 		});
 	}
 
+	/**
+	 * Handles the item key down event
+	 * @param {!Event} event
+	 * @protected
+	 */
+	_handleItemKeyDown(event) {
```

Comment:
hey @carloslancha should not we ignore the focus when an item is disabled?

Reply:
Yes @matuzalemsteles but we can iterate over it. Could you please open an issue? Thx!